### PR TITLE
feat(recommendations-plugin): Add the documentation for the recommendations plug-in

### PR DESCRIPTION
**Pull request by benoitf**
> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
Add the documentation for the recommendations plug-in

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18274

### Specify the version of the product this PR applies to.
7.27

### PR Checklist

As the author of this Pull Request I made sure that:

- [x] `vale` has been run successfully against the PR branch
- [x] Link checker has been run successfully against the PR branch
- [x] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [x] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [x] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [x] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

Change-Id: I2e6a708849d821ed1dd5ab2f9c9db9472b207c61
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

File: modules/end-user-guide/partials/proc_adding-the-vs-code-extension-using-recommendations.adoc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+.Prerequisites
+
+* A running instance of {prod-short}. To install an instance of {prod-short}, see {link-installing-an-instance}.
+* Featured VS Code extensions are available in the {prod-short} plug-in registry.
+
+.Procedure
+
+Open a workspace without any existing devfile using the *{prod-short} dashboard*:
+
+Recommendations plug-in will scan files, discover languages and install VS Code extensions matching these languages.
```
Comment (ericwill) [inline modules/end-user-guide/partials/proc_adding-the-vs-code-extension-using-recommendations.adoc]:
```suggestion
The recommendations plug-in will scan files, discover languages and install VS Code extensions matching these languages.
```

File: modules/end-user-guide/partials/proc_adding-the-vs-code-extension-using-recommendations.adoc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Disable this feature by setting `extensions.ignoreRecommendations` to true in the devfile attributes.
+
+Recommendations plug-in can suggest VS Code extensions to install when opening files. It suggests extensions allowing to work with the given files.
```
Comment (ericwill) [inline modules/end-user-guide/partials/proc_adding-the-vs-code-extension-using-recommendations.adoc]:
```suggestion
The recommendations plug-in can suggest VS Code extensions to install when opening files. It suggests extensions based on the workspace content, allowing the user to work with the given files.
```

**Review by ericwill — APPROVED** [review]

**Review by tolusha — APPROVED** [review]

**Review by yhontyk — APPROVED** [review]
LGTM. This should be merged after the 7.27 release since we publish from master the current version.

**Comment by themr0c** [comment]
7.27 is out, we can merge.

**Pull request merged**